compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/WebSocketServiceValidator.java

```diff
@@ -39,7 +41,11 @@ public void validate() {
         classDefNode.members().stream().filter(child -> child.kind() == SyntaxKind.OBJECT_METHOD_DEFINITION
                 || child.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION).forEach(methodNode -> {
             FunctionDefinitionNode functionDefinitionNode = (FunctionDefinitionNode) methodNode;
-            if (functionDefinitionNode.qualifierList().get(0).text().equals(PluginConstants.REMOTE_KEY_WORD)) {
+            NodeList<Token> qualifierList = functionDefinitionNode.qualifierList();
+            if (qualifierList.isEmpty()) {
+                return;
+            }
+            if (qualifierList.get(0).text().equals(PluginConstants.REMOTE_KEY_WORD)) {
```

Comment:
`qualifierList.get(0)` is used multiple times, better if we could create a variable to hold this and use that particular variable in every other place.